daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto

```diff
@@ -127,3 +127,19 @@ message NodeRollback { // *since version 14*
 }
 
 // architecture-handbook-entry-end: Nodes
+
+message Versioned {
+  string version = 1;
+  bytes payload = 2;
+}
+
+message FatContractInstance {
+  bytes contract_id = 1;
+  com.daml.lf.value.Identifier template_id = 2;
+  bytes create_arg = 3;
```

Comment:
Add comment as to how this is encoded. I assume it's encoding is determined by the version around the payload?